.github/actions/setup-rust/action.yml

```diff
@@ -34,25 +34,52 @@ runs:
   using: "composite"
   steps:
     - uses: mozilla-actions/sccache-action@7d986dd989559c6ecdb630a3fd2557667be217ad # v0.0.9
-    - run: |
-        echo "RUSTC_WRAPPER=$SCCACHE_PATH" >> $GITHUB_ENV
-        echo "SCCACHE_GHA_ENABLED=true" >> $GITHUB_ENV
-      shell: bash
-    - name: Disable Windows Defender
-      if: ${{ runner.os == 'Windows' }}
-      run: Set-MpPreference -DisableRealtimeMonitoring $true
-      shell: powershell
 
+    # Extract Rust version early so we can use it in the cache key
     - name: Extract Rust version
       run: |
         RUST_TOOLCHAIN=$(grep 'channel' rust-toolchain.toml | awk -F '"' '{print $2}')
         echo "RUST_TOOLCHAIN=$RUST_TOOLCHAIN" >> $GITHUB_ENV
       shell: bash
       working-directory: ./rust
+
+    # Build a cache key
+    - name: Compute sccache cache key
+      id: sccache_keys
+      shell: bash
+      run: echo "key=sccache-${{ runner.os }}-${{ env.RUST_TOOLCHAIN }}" >> "$GITHUB_OUTPUT"
+
+    # PRs: read-only (no uploads)
+    - name: Configure sccache (PR, read-only)
+      if: ${{ github.ref_name != 'main' }}
+      shell: bash
+      run: |
+        echo "SCCACHE_GHA_ENABLED=true" >> $GITHUB_ENV
```

Comment:
This is not read-only, is it?

Reply:
Another option is to see if Azure blob storage is a supported sccache backend.

Reply:
Yeah we'll need to just store the rust cache in Azure

https://github.com/mozilla/sccache/blob/main/docs/Azure.md